test/metric/prometheus.go

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+)
+
+type PrometheusMetricValueFetcher struct {
+	baseMetricValueFetcher
+}
+
+var _ MetricValueFetcher = (*PrometheusMetricValueFetcher)(nil)
+
+func (f *PrometheusMetricValueFetcher) Fetch(namespace, metricName string, stat Statistics) (MetricValues, error) {
+	dimensions := f.getMetricSpecificDimensions(metricName)
+	values, err := f.fetch(namespace, metricName, dimensions, stat)
+	if err != nil {
+		log.Printf("Error while fetching metric value for %v: %v", metricName, err.Error())
+	}
+	return values, err
+}
+
+func (f *PrometheusMetricValueFetcher) isApplicable(metricName string) bool {
+	prometheusSupportedMetric := f.getPluginSupportedMetric()
+	_, exists := prometheusSupportedMetric[metricName]
+	return exists
+}
+
+func (f *PrometheusMetricValueFetcher) getPluginSupportedMetric() map[string]struct{} {
+	return map[string]struct{}{
+		"prometheus_test_counter":       {},
```

Comment:
So we would not do histogram or untyped?

Reply:
We need to enhance the framework to account of metrics we dont expect. Currently we only handle expected metrics.
Histogram & untyped would be ones we dont expect.

Reply:
Weird. I forgot to update my comments. Yes agree on that. If yes, it would be better to have document on what type of Prometheus metrics type we support though.

Reply:
Ill create a task for tracking the enhancement. By document, did you mean just add a comment here? If yes, I just added a comment.